controller/program/src/instructions/protocol_fee/set_protocol_fee.rs

```diff
@@ -37,51 +38,31 @@ pub fn set_protocol_fee_checked<'acc>(
 
     verify_signers(abr, &accs.0, &SET_PROTOCOL_FEE_IX_IS_SIGNER.0)?;
 
-    verify_not_rebalancing_and_not_disabled(pool)?;
+    verify_not_rebalancing_and_not_disabled_v2(pool)?;
 
-    let SetProtocolFeeIxArgs {
-        trading_bps,
-        lp_bps,
-    } = SetProtocolFeeIxData::parse_no_discm(ix_data_no_discm).ok_or(INVALID_INSTRUCTION_DATA)?;
+    let protocol_fee_nanos = SetProtocolFeeIxData::parse_no_discm(
+        ix_data_no_discm
+            .first_chunk()
+            .ok_or(INVALID_INSTRUCTION_DATA)?,
+    );
 
-    [trading_bps, lp_bps]
-        .into_iter()
-        .try_for_each(|bps| bps.map_or_else(|| Ok(()), verify_valid_fee_bps))?;
+    FeeNanos::new(protocol_fee_nanos).map_err(|_| Inf1CtlCustomProgErr(Inf1CtlErr::FeeTooHigh))?;
 
-    Ok((
-        accs,
-        SetProtocolFeeIxArgs {
-            trading_bps,
-            lp_bps,
-        },
-    ))
+    Ok((accs, protocol_fee_nanos))
 }
 
 #[inline]
 pub fn process_set_protocol_fee(
     abr: &mut Abr,
     accs: &SetProtocolFeeIxAccounts,
-    SetProtocolFeeIxArgs {
-        trading_bps,
-        lp_bps,
-    }: &SetProtocolFeeIxArgs,
+    protocol_fee_nanos: u32,
```

Comment:
nit: in general once we've parsed/verified something, we want to use the verified type: making this change ensures that whatever `protocol_fee_nanos` arg we call `process_set_protocol_fee` with has been verified to be a valid `FeeNanos`

```suggestion
    protocol_fee_nanos: FeeNanos, // then need to use `*pool_protocol_fee_nanos = protocol_fee_nanos.get();` below
```